pkg/util/error.go

```diff
@@ -22,8 +27,35 @@ const (
 
 // WriteError write a go error with the correct status code.
 func WriteError(err error, w http.ResponseWriter) {
+	var connectErr *connect.Error
+	if ok := errors.As(err, &connectErr); ok {
+		writeErr := errorWriter.Write(w, &http.Request{
+			Header: http.Header{"Content-Type": []string{"application/json"}},
+		}, err)
+		if writeErr != nil {
+			http.Error(w, writeErr.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
 	status, cerr := ClientHTTPStatusAndError(err)
-	http.Error(w, cerr.Error(), status)
+	WriteErrorWithStatus(cerr, w, status)
+}
+
+func WriteErrorWithStatus(err error, w http.ResponseWriter, status int) {
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
I don't think you want to Set `Content-Type` twice. There are also cases where headers have been set before, so maybe we only want to set something if nothing is set:

https://github.com/grafana/pyroscope/blob/dfc13514e978c69f63ca347d1fc5d266e7adb852/pkg/querier/http.go#L66-L69

I also remember for some browsers you need `text/plain` to get them to show the text as monospaced font or something.